name: Validate URLs

on:
  workflow_dispatch:
  schedule:
    - cron: '0 13 * * 1'

jobs:

  validate_urls:
    name: Validate URLs
    runs-on: ubuntu-latest
    environment: 'production'
    permissions:
      contents: read
      issues: write
    steps:
      - uses: actions/checkout@08c6903cd8c0fde910a37f88322edcfb5dd907a8 # v5.0.0

      - name: Setup Python
        uses: actions/setup-python@e797f83bcb11b83ae66e0230d6156d7c80228e7c # v6.0.0
        with:
          python-version: 3.9
          architecture: x64

      - uses: astral-sh/setup-uv@557e51de59eb14aaaba2ed9621916900a91d50c6 # v6.6.1

      - name: Get invalid URLs
        id: urls
        run: |
          output=$(uv run python utility_scripts/plugin_definitions/check_links.py)
          echo "INVALID_URL<<EOF" >> $GITHUB_ENV
          echo "$output" >> $GITHUB_ENV
          echo "EOF" >> $GITHUB_ENV
      - uses: JasonEtco/create-an-issue@1b14a70e4d8dc185e5cc76d3bec9eab20257b2c5 # v2.9.2
        if: env.INVALID_URL != ''
        env:
          GITHUB_TOKEN: ${{ secrets.MELTYBOT_GITHUB_AUTH_TOKEN }}
          INVALID_URLS: ${{ env.INVALID_URL }}
        with:
          filename: .github/ISSUE_TEMPLATE/invalid_urls.md
          update_existing: true
